Feat: Add `resolveLinks` option to improve MultilinkStoryblok

RKcode opened the pull request:
Handle [this options](https://www.storyblok.com/docs/guide/in-depth/rendering-the-link-field#resolving-a-link-to-a-story-with-url) from storyblok.

The code only handle the part were `resolve_links` value is `story`. In this PR, I wanted to also handle `url` and `link` value.
I think most of Storyblok's dev resolve with `url` value instead, now we have all three choices.

By default, I put `story` value to the parameter to avoid breaking changes, but I think it's not optimal. When you don't put `resolve_links` in your code, the link object won't have any object.

dohomi commented:
Hi @RKcode thanks for your contribution, one thought I have about this: doesn't resolveLinks is an option you would switch on/off per specific API endpoint? This means, sometimes you'd have them defined to url sometimes to story.. this would make it incomplete to have only one settings for it. Currently I solve this inside my component settings and extend the config for the extended usecase

RKcode commented:
Hey @dohomi !
You're absolutely right, it's a param when you fetch stories, so you can change it from a fetch to another. But I think, most of us will only have one component/function to handle Storyblok's Link, and probably with `url` or `link` (less rate limit).

Maybe, the resolveLinks option can be an Array of option instead, WDYT ?

At least, this gives a choice for people that use only one of those options in their code. Note that today, it only handles the resolveLinks `story`, it doesn't handle the default option : the no resolve links param, type: `story: undefined`.

Extend is a great solution anyway, especially when you resolveLinks differently 👍

dohomi commented:
I think the best would be to support an array of all available types and guide the dev how to extend their own typings with either of the options within their code? Similar like how I tried to give pointers in https://github.com/dohomi/storyblok-generate-ts?tab=readme-ov-file#resolve-relations

RKcode commented:
Hey @dohomi,
I've changed the `resolveLinks` param to be an array, so I also changed the CLI option to handle multiple options separated with commas 👍 
If you don't pass any option, `story` will not be defined, as it should be when you don't have `resolve_links` in your fetch request:
![Screenshot 2024-04-08 at 17 36 42](https://github.com/dohomi/storyblok-generate-ts/assets/12799098/dca53b15-bb78-4467-afbe-233778c59cd3)
Here is an another example: `resolveLinks=url,story`
![Screenshot 2024-04-08 at 17 38 26](https://github.com/dohomi/storyblok-generate-ts/assets/12799098/94e2132f-6116-4517-9d22-9ecf7207f8f5)

I know that we can still extend type in another file but an option inside the lib is better for me !

dohomi commented:
released in 2.1.0 👍 thanks for your contribution
